### PR TITLE
New package: gazou-0.3.3

### LINKED ISSUE #48156: Package request: Gazou
### Package name

gazou

### Package homepage

https://github.com/kamui-fin/gazou

### Description

Gazou is a Japanese and Chinese OCR application written in C++. It utilizes image processing, and other parameters to improve the accuracy.

### Does the requested package meet the package requirements?

System, Compiled

### Is the requested package released?

Yes

### PR DESCRIPTION
**Pull request by meator**
@jaminW55 Thanks for writing [the basic template](https://github.com/void-linux/void-packages/issues/48156#issuecomment-1885404699), I have used it to create this. Would you mind testing this pull request? I can give you instructions on how to do so if you want to.

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

closes #48156

### CODE REVIEW

**Comment by jaminW55** [comment]
Hello. That's great!  Thanks very much!

This is my first time, how could I test this pull request!?

**Comment by meator** [comment]
Since this PR has only a single template, the simplest way to compile it is to copy the template (https://raw.githubusercontent.com/meator/void-packages/gazou/srcpkgs/gazou/template) to `srcpkgs/gazou/template` and then do `./xbps-src pkg gazou` (assuming you have set up xbps-src according to [Quick start](https://github.com/void-linux/void-packages?tab=readme-ov-file#quick-start)).

**Comment by jaminW55** [comment]
Excellent News. I just installed using the provided template and it ran without issue.

I'm still trying to ascertain where I went wrong as the build instructions look fairly different from my own, but I'm very happy it is working!

Thanks for your help!

**Comment by meator** [comment]
> I'm still trying to ascertain where I went wrong as the build instructions look fairly different from my own, but I'm very happy it is working!

```
build_style=cmake
```
Handles all building and instalation. You usually don't need to specify `do_configure()`, `do_build()` and `do_install()` (unless the build process is nonstandard in some way).

**Comment by jaminW55** [comment]
> ```
> build_style=cmake
> ```
> 
> Handles all building and instalation. You usually don't need to specify `do_configure()`, `do_build()` and `do_install()` (unless the build process is nonstandard in some way).

Is this generally true for build instructions more broadly? The extra instructions are just for intermitiary steps? 

There are a few other programs for which I am working on creating templates, including the Tachidesk package request which uses Java.

**Comment by jaminW55** [comment]
In terms of moving this package forward into the repos, what else do we have left in the process? 

I'd like to help expand Void's access to Japanese tools, readers, and programs more broadly.

**Comment by meator** [comment]
This isn't really a place to ask about the basics of xbps-src, but I'll try to explain it a bit anyway.

> Is this generally true for build instructions more broadly? The extra instructions are just for intermitiary steps?

The build proces has several [phases](https://github.com/void-linux/void-packages/blob/master/Manual.md#package-build-phases). They correspond to a `do_<phase>()` function in the template. Only `do_install()` is mandatory, but `do_configure()` and `do_build()` are needed most of the time. The [build styles](https://github.com/void-linux/void-packages/blob/master/Manual.md#build_scripts) provide generic definitions of these. For example `do_configure()` of `build_style=cmake` is defined here: https://github.com/void-linux/void-packages/blob/master/common/build-style/cmake.sh#L4C1-L89 It is much more complex than `cmake .` because it needs to handle Void linux specific things, cross compilation (this is an important step that gets abstracted away by the build style which is great because you don't have to deal with it), set prefix to `/usr` and more. Void linux has two libc implementations (glibc and musl) and a bunch of supported architectures. The template should support all of these (unless it can't). The build styles are written to handle this. You should prefer using predefined build styles instead of writing it from scratch (which you can do by just defining the function in the template) because you would have to do all these things yourself.

You can influence these build styles by setting some variables like `configure_args`, `make_build_args` etc. (it depends on the build style chosen). There are also `pre_<phase>()` and `post_<phase>()` functions that get executed before and after the phase runs. You can use these to do some setup, [fixup the upstream build system a bit if necessary](https://github.com/void-linux/void-packages/blob/master/srcpkgs/windowchef/template#L14-L17) or [install a license if required](https://github.com/void-linux/void-packages/blob/master/srcpkgs/windowchef/template#L19-L21)[^1]. [^2]

Then you have the variables like `version`, `short_desc` etc. They are pretty self explanatory. Read the [manual](https://github.com/void-linux/void-packages/blob/master/Manual.md) for more info.

If you're having trouble, try looking at some other templates for inspiration. And read the [manual](https://github.com/void-linux/void-packages/blob/master/Manual.md). Use `xlint` on your templates.

> In terms of moving this package forward into the repos, what else do we have left in the process?

Merging of PRs can take some time. I believe that this PR is ready to be merged, some maintainer just has to take a look at it and merge it. Merging of new packages can take longer compared to simple package update PRs. See [CONTRIBUTING](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md)

[^1]: Some licenses state that a copy of the license must be distributed alongside the program. You usually don't have to worry about it. `xlint` will warn you if your program is using one of these licenses.
[^2]: I have chosen the `windowchef` at random as an example to link to, it demonstrates the usage of these functions.

**Comment by jaminW55** [comment]
Hello. Just checking to see if this is going to be merged and if I should close out my current issue.

**Comment by jaminW55** [comment]
> > In terms of moving this package forward into the repos, what else do we have left in the process?
> 
> Merging of PRs can take some time. I believe that this PR is ready to be merged, some maintainer just has to take a look at it and merge it. Merging of new packages can take longer compared to simple package update PRs. See [CONTRIBUTING](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md)

Is it possible for me to takeover and move forward as maintainer? i think I have enough knowledge of this program and a bit more experience building void packages now that I could probably maintain it. It isn't updated too often and I'd like to get this moved to the repos.

Sorry for the delay, but any direction you could provide would be most appreciated.

**Comment by meator** [comment]
I will update this PR and include you as the maintainer if you so wish.

What is your maintainer string? (for example my is `meator <meator.dev@gmail.com>`, it should follow this format).

**Comment by jaminW55** [comment]
jaminW55 <jaminW55@proton.me> should work fine.

File: srcpkgs/gazou/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'gazou'
+pkgname=gazou
+version=0.3.3
+revision=1
+build_style=cmake
+configure_args="-DGUI=ON"
+hostmakedepends="pkg-config qt5-qmake qt5-host-tools"
+makedepends="qt5-devel tesseract-ocr-devel leptonica-devel qt5-x11extras-devel"
+short_desc="Japanese and Chinese OCR application"
+maintainer="meator <meator.dev@gmail.com>"
```
Comment (jaminW55) [inline srcpkgs/gazou/template:10]: Add "jaminW55 <jaminW55@proton.me>" as maintainer.

File: srcpkgs/gazou/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+revision=1
+build_style=cmake
+configure_args="-DGUI=ON"
+hostmakedepends="pkg-config qt5-qmake qt5-host-tools"
+makedepends="qt5-devel tesseract-ocr-devel leptonica-devel qt5-x11extras-devel"
+short_desc="Japanese and Chinese OCR application"
+maintainer="jaminW55 <jaminW55@proton.me>"
+license="GPL-3.0-only"
+homepage="https://github.com/kamui-fin/gazou"
+_QHotkey=1.5.0
```
[the remark this answers is not in the document]
Comment (meator) [reply srcpkgs/gazou/template]: Done, thanks.

[the remark this answers is not in the document]
Comment (meator) [reply srcpkgs/gazou/template]: Thanks, I have put it below `revision` and I've simplified the template a bit. I didn't notice your updated message mentioning the submodule commit, I've fixed that too now.

**Pull request merged**